spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/vectorized/parquet/TestParquetVectorizedReads.java

```diff
@@ -493,6 +496,21 @@ static Stream<Arguments> goldenFilesAndEncodings() {
                                             encoding, e.getKey(), e.getValue(), vectorized))));
   }
 
+  private File resourceUrlToLocalFile(URL url) throws IOException, URISyntaxException {
+    if ("file".equals(url.getProtocol())) {
+      return Paths.get(url.toURI()).toFile();
+    }
+
+    String name = Paths.get(url.getPath()).getFileName().toString(); // e.g., string.parquet
+    String suffix = name.contains(".") ? name.substring(name.lastIndexOf('.')) : "";
+    Path tmp = java.nio.file.Files.createTempFile("golden-", suffix);
```

Comment:
nit: we normally use the junit5 `TempDir`, which would auto deleted after the test execution.

```
@TempDir private Path temp;
...
File data = File.createTempFile(("golden-", suffix, temp.toFile());

```